utilities/bigquery.py

```diff
@@ -16,6 +18,27 @@ def get_likes(id_token):
         where
             json_value(data, '$.uid') = '{uid}'
     '''
-    results = db.query(query)
+    results = db.query(query, job_config=job_config)
     likes = [json.loads(row[0]) for row in results]
     return likes
+
+def get_song(id_token, time_of_day, likes):
+    """Get song recommendation for the logged in user."""
+    uid = auth.verify_id_token(id_token)['uid'] # pylint: disable=unused-variable
+    like_ids = ','.join(
+        [str(like['id']) for like in likes if like['time_of_day'] == time_of_day]
+    )
+    likes_string = f'\nand id not in ({like_ids})' if like_ids else ''
+    query = f'''
+        select
+            *
+        from
+            `faf-starter.noon.recommendations`
+        where
+            time_of_day = '{time_of_day}'{likes_string}
```

Comment:
I typically handle this pattern by making a list of conditions and joining on AND:

```py
where_conditions = [
  f'time_of_day = '{time_of_day}',
]

if like_ids:
  formatted_like_ids = ','.join(
      str(like['id'])
      for like in likes
      if like['time_of_day'] == time_of_day)
  where_conditions.append(f'id not in ({formatted_like_ids})')

where_clause = ' AND '.join(where_conditions)

query = f'''
  select * from `faf-starter.noon.recommendations`
  where {where_conditions}
'''
```

Having a list of conditions makes it easier to deal with the case where the where clause is empty or if you didn't have a constant time_of_day query